src/monfaction.h

```diff
@@ -0,0 +1,41 @@
+#ifndef MONFACTION_H
+#define MONFACTION_H
+
+#include "string_id.h"
+#include "int_id.h"
+#include <unordered_map>
+
+class monfaction;
+class JsonObject;
+
+using mfaction_id = int_id<monfaction>;
+using mfaction_str_id = string_id<monfaction>;
+
+enum mf_attitude {
+    MFA_BY_MOOD = 0,    // Hostile if angry
+    MFA_NEUTRAL,        // Neutral even when angry
+    MFA_FRIENDLY        // Friendly
+};
+
+typedef std::unordered_map< mfaction_id, mf_attitude > mfaction_att_map;
+
+namespace monfactions
+{
+void finalize_monfactions();
+void load_monster_faction(JsonObject &jo);
+mfaction_id get_or_add_faction( const std::string &name );
+}
```

Comment:
A namespace is an interesting solution. I usually put this stuff as static functions into the classes that it refers to (but this here will work, too).

However, I really don't like the pointless repetition of the name of the name of the finalized thing: `monfactions::finalize_monfactions`. This is only called once (exactly like this) from init.cpp (and should never be called from anywhere else. It should be pretty much clear, what is finalized. `monfactions::finalize` should be enough.